### ISSUE TITLE
No newline emitted at the end of the document in replace mode

### ISSUE THREAD

**Issue by jonathanio** [terraform-docs/terraform-docs#465]
### Describe the bug

When running `terraform-docs` (as tested with 0.12.0 and 0.12.1) in `replace` mode, and regardless of if `footer-from` is set or not, the resulting `README.md` is created without a newline at the end.

If no `footer-from` is set, then the final table (i.e. _Outputs_) does not finish with a newline character on its last row. When using `footer-from`, there is no newline at the end of the document even when the sourced file (e.g. `.footer.md` as per the configuration below) contains a newline at the end of it.

This manifests itself for me mainly where I am using `pre-commit` on a repository, as this inevitably leads to a fight between `end-of-line-fixer` from [`pre-commit-hooks`](https://github.com/pre-commit/pre-commit-hooks) and the output of this program when changes are made to Terraform files.

### How can we reproduce it?

```sh
# terraform-docs --config .terraform-docs.yaml .
```

```yaml
formatter: markdown

header-from: .header.md
footer-from: .footer.md

sections:
  show-all: true

sort:
  enabled: true
  by:
    - required

output:
  file: README.md
  mode: replace
  template: |-
    {{ .Content }}
```

### Environment information

```
terraform-docs version 0.12.1 4714dd3 linux/amd64 BuildDate: 2021-03-29T17:07:19Z
```

**Comment by jonathanio** [comment]
It also breaks the rule `MD047` if you're using `markdownlint` as well, as I found out when I disabled the end-of-file checker in `pre-commit` to stop these tools fighting. :smile:

**Comment by jonathanio** [comment]
This does work for me. I'll close this as we have a way to handle this within the templating engine.

**Comment by BMeach** [comment]
For anyone with the same issue, the `{{- printf "\n" -}}` fix still works on v0.17.0

**Comment by neiser** [comment]
However, when combined with [the workaround for AsciiDoc here](https://github.com/terraform-docs/terraform-docs/issues/580), it does not seem to work, as otherwise [this check](https://github.com/terraform-docs/terraform-docs/blob/84f5f685127bcacaa9f6cf1a77cad365f7d64a8b/print/config.go#L306) will trigger an error:

> Error: value of '--output-template' is missing end comment

I've tried both variants:

```
output:
  template: |-
    // BEGIN_TF_DOCS
    {{ .Content }}
    // END_TF_DOCS {{- printf "\n" -}}
```

and

```
output:
  template: |-
    // BEGIN_TF_DOCS
    {{ .Content }}
    // END_TF_DOCS 
    {{- printf "\n" -}}
```

@jonathanio @khos2ow I think that requires a fix in `validate`?
